force-app/main/default/lwc/lookup/lookup.js

```diff
@@ -1,287 +1,320 @@
 import { LightningElement, track, api } from "lwc";
 
 const MINIMAL_SEARCH_TERM_LENGTH = 2; // Min number of chars required to search
-const SEARCH_DELAY = 300; // Wait 300 ms after user stops typing then, peform search
-
+const SEARCH_DELAY = 300; // Wait 300 ms after user stops typing then, perform search
+
+// TODO: Localize with Custom Labels.
+const SELECT_ICON_ALTERNATIVE_TEXT = "Selected item icon";
+const LOADING_LABEL = "Loading";
+const SEARCH_ICON_ALTERNATIVE_TEXT = "Search icon";
+const REMOVE_BUTTON_ALTERNATIVE_TEXT = "Remove selected option";
+const RESULT_ICON_ALTERNATIVE_TEXT = "Result item icon";
+const MULTI_ENTRY_ARIA_LABEL = "Selected Options:";
 export default class Lookup extends LightningElement {
-  @api label;
-  @api selection = [];
-  @api placeholder = "";
-  @api isMultiEntry = false;
-  @api errors = [];
-  @api scrollAfterNItems;
-  @api customKey;
-  @api disabled;
-
-  @track searchTerm = "";
-  @track searchResults = [];
-  @track hasFocus = false;
-
-  cleanSearchTerm;
-  blurTimeout;
-  searchThrottlingTimeout;
-
-  // EXPOSED FUNCTIONS
-
-  @api
-  setSearchResults(results) {
-    this.searchResults = results.map(result => {
-      // Clone and complete search result if icon is missing
-      if (typeof result.icon === "undefined") {
-        const { id, sObjectType, title, subtitle } = result;
-        return {
-          id,
-          sObjectType,
-          icon: "standard:default",
-          title,
-          subtitle
-        };
-      }
-      return result;
-    });
-  }
-
-  @api
-  getSelection() {
-    return this.selection;
-  }
-
-  @api
-  getkey() {
-    return this.customKey;
-  }
-
-  // INTERNAL FUNCTIONS
-
-  updateSearchTerm(newSearchTerm) {
-    this.searchTerm = newSearchTerm;
-
-    // Compare clean new search term with current one and abort if identical
-    const newCleanSearchTerm = newSearchTerm
-      .trim()
-      .replace(/\*/g, "")
-      .toLowerCase();
-    if (this.cleanSearchTerm === newCleanSearchTerm) {
-      return;
+    labels = {
+        selectIcon: {
+            alternativeText: SELECT_ICON_ALTERNATIVE_TEXT,
+        },
+        loading: {
+            assistiveText: LOADING_LABEL,
+        },
+        searchIcon: {
+            alternativeText: SEARCH_ICON_ALTERNATIVE_TEXT,
+        },
+        removeButton: {
+            alternativeText: REMOVE_BUTTON_ALTERNATIVE_TEXT,
+        },
+        resultIcon: {
+            alternativeText: RESULT_ICON_ALTERNATIVE_TEXT,
+        },
+        multiEntry: {
+            ariaLabel: MULTI_ENTRY_ARIA_LABEL,
+        },
+    };
+
+    _selection = [];
+
+    @api
+    get selection() {
```

Comment:
selection was originally being modified and lwc/eslint didn't like an `@api` being directly modified.  Change to having `@api` getter/setter modify a private variable.